pkg/gui/staging_panel.go

```diff
@@ -174,3 +174,54 @@ func (gui *Gui) HandleOpenFile() error {
 
 	return gui.helpers.Files.OpenFile(file.GetPath())
 }
+
+func (gui *Gui) handleEditHunk() error {
+	return gui.withLBLActiveCheck(func(state *LblPanelState) error {
+		return gui.editHunk(state.SecondaryFocused, state)
+	})
+}
+
+func (gui *Gui) editHunk(reverse bool, state *LblPanelState) error {
+	file := gui.getSelectedFile()
+	if file == nil {
+		return nil
+	}
+
+	hunk := state.CurrentHunk()
+	patchText := patch.ModifiedPatchForRange(gui.Log, file.Name, state.GetDiff(), hunk.FirstLineIdx, hunk.LastLineIdx(), reverse, false)
+	patchFilepath, err := gui.git.WorkingTree.SaveTemporaryPatch(patchText)
+	if err != nil {
+		return err
+	}
+
+	lineOffset := 3
```

Comment:
is this the number of context lines? Because if so, we should make it dynamic based on whether the user has increased/decreased the amount of context lines (via '{' and '}').

Reply:
`lineOffset` is the number of header lines of patches (like ↓), which is always 3 lines.

```diff
--- a/README.md
+++ b/README.md
@@ -297,11 +297,12 @@ If you would like to support the development of lazygit, consider [sponsoring me
```

Reply:
Will it crash if you try to edit a hunk on a binary file?

Reply:
@mark2185 For binary files, `edit hunk` cannot be performed in the first place because the staging panel cannot be opened.